src/ImageManipulation.php

```diff
@@ -1057,4 +1069,204 @@ protected function castDimension($value, $dimension)
         }
         return $value;
     }
+
+    /**
+     * Create an equivalent DBFile object
+     * @return AssetContainer
+     */
+    private function copyImageBackend(): AssetContainer
+    {
+        // Store result in new DBFile instance
+        /** @var DBFile $file */
+        $file = DBField::create_field(
+            'DBFile',
+            [
+                'Filename' => $this->getFilename(),
+                'Hash' => $this->getHash(),
+                'Variant' => $this->getVariant()
+            ]
+        );
+
+        $backend = $this->getImageBackend();
+        if ($backend) {
+            $file->setImageBackend($backend);
+        }
+
+        return $file->setOriginal($this);
+    }
+
+    /**
+     * Add a HTML attribute when rendering this object. This method is immutable, it will return you a copy of the
+     * original object.
+     * @param string $name
+     * @param mixed $value
+     * @return AssetContainer
+     */
+    public function setAttribute($name, $value)
+    {
+        $file = $this->copyImageBackend();
+        $file->initAttributes(array_merge(
+            $this->attributes,
+            [$name => $value]
+        ));
+
+        return $file;
+    }
+
+    /**
+     * Initialise the attirbutes on this object. Should only be called imiditely after intialising the object.
+     * @param array $attributes
+     * @internal
+     */
+    public function initAttributes(array $attributes)
```

Comment:
Does this need to be a public function?  Seems like this should be a private?

Reply:
I'll double check to be sure, but I think a trait cannot access its own private members on an other class.

Reply:
Just tried running the unit test locally with private/protected and it fails both time. Basically, `Image` can't called a private/protected method on 'DBFile` even if they share a trait.

I marked the method `@internal` to communicate to third party devs they shouldn't be calling this method directly.

Technically, this will only ever be called on DBFile and `Image` doesn't need it. We could move the method into `DBFile` if you think that makes more sense.